UPGRADE.md

```diff
@@ -1,10 +1,41 @@
 # Upgrade Guide
 
-_After `7.4.0` all `MINOR` and `MAJOR` version bumps will have upgrade notes 
-posted here._
+_`MAJOR` version bumps will have upgrade notes posted here._
+
+[2020-09-28] 7.x.x to 8.x.x
+-----------------------------
+
+### Overview
+
+Version `8.x.x` is the first version that officially drops support for Java 7. 
+
+- Removal of dependencies offering functionality included in Java 8 and beyond:
+    - [Guava functional interfaces](https://github.com/twilio/twilio-java/pull/574), for example:
+        - `com.google.common.base.Joiner` -> `String.join()`
+        - `com.google.common.collect.Lists` -> `Array.asList()`
```

Comment:
Only changes that affect client usage should be listed here. For instance, it doesn't matter to the client that we're using `String.join()` under the hood instead of a Guava joiner. It's an implementation detail. Things that _do_ matter to consumers is the date stuff, removal of `Range` support, removal of deprecated stuff, and maybe some of the concurrency interfaces.